### DISCUSSION TITLE
JBD AP20S003 BMS ...CAN and RS485, inverters are Rich Solar H-3000  48v

### DISCUSSION THREAD

**Discussion by CapnGimp** [ai-republic/bms-to-inverter#31] [Q&A]
I have a JBD AP20S003 BMS uses either of  bluetooth CAN and RS485, my inverters are Rich Solar H-3000  48v units. Have a Pi3B+ I am  now experimenting with and will get an expansion HAT 2chan canbus quick as I see you have it supported. Been watching the thread at DIY Solar for a while. Thanks a BUNCH! CapnGimp on forums, will update your thread.

**Comment by ai-republic** [comment]
OK, nice that you like it! 
If you have any questions don't hesitate to reach out to me :)
Do you have a CAN communication protocol of the JBD  BMS?

**Reply by CapnGimp** [reply]
No I do not but I will search for it today and answer you here quick as I can locate it.

**Reply by CapnGimp** [reply]
is this what you are looking for? https://cdn.shopifycdn.net/s/files/1/0606/5199/5298/files/JDB_RS485-RS232-UART-Bluetooth-Communication_Protocol.pdf?v=1682577935   it is a pdf file 6 pages long with charts  first line is ( it is safe to download)  ....   Software board general protocol V4
One, physical interface
This protocol supports RS485/RS232/UART interface of Jiabaida software boardGeneral protocol, consistent with the host computer protocol, The
baud rate is 9600BPSOr other customer customized rates

**Comment by CapnGimp** [comment]
Here is a link to someone who wrote code like what you are doing, posted on github, just saw this haven't read it yet, could be sime good info for you...    https://diysolarforum.com/threads/jbd-overkill-bms-monitoring.21817/

**Reply by ai-republic** [reply]
Thanks for the RS485 communication specification. That's enough for me to implement a RS485 binding. I just added the BYD and PACE BMS CAN bindings, so the JBD RS485 will be next :)

**Comment by ai-republic** [comment]
I have added the preliminary JBD support (currently without alerts). Could you test it and see if you get data from the JBD?
Please use the _Configurator_ application and choose the JBD_RS485 binding as the BMS and NONE for inverter. Pleae also set the log level to debug. Thanks :)

**Reply by CapnGimp** [reply]
Thank you so much! I will order the CAN BUS for my Pi3B+ today and let you know the results ASAP.

**Comment by ai-republic** [comment]
@CapnGimp any progress yet? How did it go?
